### ISSUE TITLE
hadolint had missed the warning `unsupported modifier (=) in substitution`

### ISSUE THREAD

**Issue by IceCodeNew** [hadolint/hadolint#822]
- [ ] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior
hadolint should warn the invalid instruction of the following example.

### Actual behavior
hadolint had missed the warning `unsupported modifier (=) in substitution`.

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```
Should be the latest release. I actually am using the online tool `https://hadolint.github.io/hadolint/`.
```

**Dockerfile** (if relevant)

```Dockerfile
FROM scratch AS assets
COPY something_to_be_copy /some_directory/


FROM gcr.io/distroless/static:debug
COPY --from=assets / /

ARG commit_id
ARG app_version
ENV COMMIT_ID=${commit_id:=dev}
ENV APP_VERSION=${app_version:=dev}

ENTRYPOINT ["/some_directory/something_to_be_copy"]
```

**Additional environment details (OS, stack version, etc.)**

### error message:
```
Step 11/14 : ENV COMMIT_ID=${commit_id:=dev}
failed to process "${commit_id:=dev}": unsupported modifier (=) in substitution
```

`$ docker version`
```
Client:
 Version:           20.10.7
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        f0df350
 Built:             Wed Jun  2 11:51:04 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.7
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       b0f5bc3
  Built:            Wed Jun  2 11:55:29 2021
  OS/Arch:          linux/amd64
  Experimental:     true
 containerd:
  Version:          v1.4.6
  GitCommit:        d71fcd7d8303cbf684402823e425e9dd2e99285d
 runc:
  Version:          1.0.0
  GitCommit:        v1.0.0-0-g84113eef
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Comment by IceCodeNew** [comment]
hadolint should also warn if users are giving the instructions as follows:
```
ENV COMMIT_ID=${commit_id:dev}
```

It seems `docs.docker.com` failed to make the statement that
```
ARG CONT_IMG_VER
ENV CONT_IMG_VER=${CONT_IMG_VER:-v1.0.0}
```
is the only allowed way to do the trick.

(Refer to: https://docs.docker.com/engine/reference/builder/#using-arg-variables )
